pkg/providers/tinkerbell/upgrade.go

```diff
@@ -134,7 +134,23 @@ func (p *Provider) SetupAndValidateUpgradeCluster(ctx context.Context, cluster *
 		return err
 	}
 
-	return p.validateAvailableHardwareForUpgrade(ctx, currentClusterSpec, clusterSpec)
+	if err := p.validateAvailableHardwareForUpgrade(ctx, currentClusterSpec, clusterSpec); err != nil {
```

Comment:
Should this be done after `if p.clusterConfig.IsManaged()` block? Once the new hardware is applied on to existing management cluster, we would want to validate if we have enough hardware for upgrade right?

Reply:
We validate all the hardwares (hardware object) before applying it to the cluster https://github.com/aws/eks-anywhere/blob/main/pkg/providers/tinkerbell/upgrade.go#L75. I also felt the same but during normal create/upgrade too, we apply all the hardwares during PostBootstrapSetupCreate or PostBootstrapSetupUpgrade step after validating it.

Reply:
Oh i see. So the validate only runs against the catalogue object and not the cluster itself. So doing it before would be fine 👍